### PR TITLE
Remove moongen.

### LINKED ISSUE #1183: Is moongen obsolete?
I don't see any results since 2017: should we get rid of `pbench-moongen`?

### PR DESCRIPTION
**Pull request by ndokos**
Fixes #1183.

Remove pbench-moongen, its postprocessing, any mock artifacts related
to it and any gold files.

Modify the unitests script to remove it.
Modify the test-benchmark-clis script to remove it and modify the
expected output from that.

### CODE REVIEW

File: agent/bench-scripts/gold/test-benchmark-clis/test-CL.txt [hunk 1]
```diff
@@ -1841,19 +1691,13 @@ Usage: pbench-user-benchmark [options] -- <script to run>
...
-/var/tmp/pbench-test-bench/pbench-agent/pbench.log:[warn][1900-01-01T00:00:00.000000] Please switch ASAP to pbench-trafficgen, which provides MoonGen and TRex based traffic generators.
-/var/tmp/pbench-test-bench/pbench-agent/pbench.log:[warn][1900-01-01T00:00:00.000000] This script (pbench-moongen) is deprecated and will be removed in a future version of pbench (v0.61).
-/var/tmp/pbench-test-bench/pbench-agent/pbench.log:[warn][1900-01-01T00:00:00.000000] Please switch ASAP to pbench-trafficgen, which provides MoonGen and TRex based traffic generators.
-/var/tmp/pbench-test-bench/pbench-agent/pbench.log:[warn][1900-01-01T00:00:00.000000] This script (pbench-moongen) is deprecated and will be removed in a future version of pbench (v0.61).
-/var/tmp/pbench-test-bench/pbench-agent/pbench.log:[warn][1900-01-01T00:00:00.000000] Please switch ASAP to pbench-trafficgen, which provides MoonGen and TRex based traffic generators.
+grep: /var/tmp/pbench-test-bench/pbench-agent/pbench.log: No such file or directory
```
Comment (k-rister) [inline agent/bench-scripts/gold/test-benchmark-clis/test-CL.txt]: This seems peculiar, where did this come from?

Comment (ndokos) [reply agent/bench-scripts/gold/test-benchmark-clis/test-CL.txt]: I think that the deprecation notice that you added in PR #1190 ended up in that file (and was the only thing that did so). I'll see if I can rid of the error message.

Comment (portante) [reply agent/bench-scripts/gold/test-benchmark-clis/test-CL.txt]: If we merge #1254, we'll pick up the change that removes all of those useless "pbench.log: No such file or directory" error messages in the gold files.

File: agent/bench-scripts/unittests [hunk 2]
```diff
@@ -331,7 +326,6 @@ done
 
 # Final clean up, if there are no failures, we'll be able to remove the
 # hierarchy entirely.
-rm -rf $_testroot/opt $moongen_dir
```
Comment (portante) [inline agent/bench-scripts/unittests]: Should we keep the removal of `$_testroot/opt`?

Comment (ndokos) [reply agent/bench-scripts/unittests]: Added it back - got too hatchet-happy there...

File: agent/bench-scripts/gold/test-benchmark-clis/test-CL.txt (hunk 1 above)
Comment (ndokos) [reply agent/bench-scripts/gold/test-benchmark-clis/test-CL.txt]: Sounds good to me.

**Comment by portante** [comment]
The unit tests for agent and server all passed, just UMI unit tests that failed for the UI.  Filed issue #1275 to cover that.

We can merge this at any time.  If we want to wait for #1254, that is fine, too.

**Pull request merged**